entity-server-api/src/main/java/org/terracotta/entity/ClientDescriptor.java

```diff
@@ -18,11 +18,20 @@
  */
 package org.terracotta.entity;
 
+import java.io.Serializable;
+
 /**
  * An opaque token representing a specific entity instance on a specific client node.
- * This is used by server-side code to specifically communicate with or track connection status of a specific client-side
- * object instance.
+ * This is used by server-side code to specifically communicate with or track connection
+ * status of a specific client-side object instance. This is used both in actives
+ * and passives in certain cases.
  * Note that implementations are expected to provide a proper equals()/hashCode().
  */
-public interface ClientDescriptor {
+public interface ClientDescriptor extends Serializable {
+  /**
+   * Does this ClientDescriptor represent an actual client endpoint,
+   * or is it merely a marker.
+   * @return
+   */
+  boolean isValid();
```

Comment:
@myronkscott I kinda hate that, as it introduces a bug path, where a mistake is made and the two are wrong.